kubernetes/istio_rules_service.go

```diff
@@ -103,7 +104,13 @@ func (in *IstioClient) getActionDetails(namespace string, istiorule string, acti
 }
 
 func (in *IstioClient) getActionHandler(namespace string, handlerType string, handlerName string, handlerChan chan<- istioResponse) {
-	result, err := in.istio.Get().Namespace(namespace).Resource(istioTypePlurals[handlerType]).SubResource(handlerName).Do().Get()
+	handlerTypePlural, ok := istioTypePlurals[handlerType]
+	if !ok {
+		log.Warningf("Handler type %s is not supported", handlerType)
+		handlerChan <- istioResponse{}
+		return
+	}
+	result, err := in.istio.Get().Namespace(namespace).Resource(handlerTypePlural).SubResource(handlerName).Do().Get()
```

Comment:
should not the error be checked and reported here? (I see you just moved these lines so the problem (if any) was already there)

Reply:
No, this is a case where there is not yet support for this type.
So, the preferred behaviour is to return an empty type.